admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/dukan_requests/DukanRequestsViewModel.kt

```diff
@@ -105,14 +112,89 @@ class DukanRequestsViewModel(
         getRequestedDukans()
     }
 
-    override fun onApproveDukanClicked(dukanId: Uuid) {}
+    override fun onApproveDukanClicked() {
+        tryToExecute(
+            callee = {
+                dukanRepository.updateDukanStatus(
+                    dukanId = currentState.selectedDukan!!.id,
+                    status = Dukan.Status.APPROVED,
+                    message = currentState.rejectReason
+                )
+            },
+            onSuccess = { onSuccessDukanApproved() },
+            onError = ::onError,
+            dispatcher = dispatcher
+        )
+    }
 
-    override fun onRejectDukanDialogClicked(dukanId: Uuid) {}
+    private fun onSuccessDukanApproved(){
+        onDukanDetailsDismissed()
+        getRequestedDukans()
+        viewModelScope.launch {
+            showSnackBar(
+                title = stringProvider.getString(Res.string.status_updated_title),
+                message = stringProvider.getString(Res.string.dukan_approved_successfully),
+                isSuccess = true
+            )
+        }
+    }
 
-    override fun onRejectDukanDialogCanceled() {}
+    override fun onRejectDukanClicked() {
+        onDukanDetailsDismissed()
+        viewModelScope.launch {
+            delay(100)
+            updateState { it.copy(isRejectDialogShown = true) }
+        }
+    }
 
-    override fun onRejectDukanConfirmed() {}
-    override fun onRejectionMessageChanged() {}
+    override fun onRejectDukanDialogDismissed() {
+        updateState {
+            it.copy(
+                isRejectDialogShown = false,
+                rejectReason = "",
+                isRejectBtnLoading = false,
+            )
+        }
+    }
```

Comment:
i think closing the dialog here is just enough, the others variables will take their values automatically, test it

```suggestion
    override fun onRejectDukanDialogDismissed() {
        updateState {
            it.copy(
                isRejectDialogShown = false
            )
        }
    }
```